galsim/utilities.py

```diff
@@ -330,34 +349,137 @@ def thin_tabulated_values(x, f, rel_err=1.e-4, preserve_range=False):
         # Nothing to do
         return x,f
 
+    total_integ = np.trapz(abs(f), x)
```

Comment:
You removed the lines that used to be after this:
```
if total_integ == 0:
    return np.array([ x[0], x[-1] ]), np.array([ f[0], f[-1] ])
```
I think they're probably a sanity check worth keeping.